### ISSUE TITLE
Tray menu spawns multiple instances which don't disappear until MagicPods is closed

### ISSUE THREAD

**Issue by stamminator** [steam3d/MagicPods-Windows#413]
When "bring taskbar to front" mode is active (which I've documented [here](https://github.com/ramensoftware/windhawk-mods/discussions/1205#discussioncomment-11175614)), the state of the MagicPods context menu gets "disconnected" from the app's state, causing multiple instances to hang around indefinitely until the MagicPods process is closed.

![Animation](https://github.com/user-attachments/assets/e8a7013a-a617-492f-9eb5-5f7846e213ac)

Having no knowledge of the code base, my hunch is that the tray menu is not being managed as a singleton. Additionally, most apps seem to have trouble when "BTTF" mode (not sure what else to call it) is active while their tray menu is opened, but the one app I've seen that handles it best is [Monitorian](https://github.com/emoacht/Monitorian). It may be worth checking out how they do event handling for their tray menu.

**Comment by steam3d** [comment]
Hi. It's not a tray menu; it's an entire window. I set the property 'TopMost=True'. The window will close when it receives a deactivated event.

Since 'TopMost' works randomly and for privacy reasons, I do not globally catch mouse clicks and keyboard presses to force the window to close when the user clicks outside of it.

**Comment by stamminator** [comment]
@steam3d If I'm understanding correctly, it sounds like there's no way to avoid at least one of these menus from being "orphaned" without globally listening to M/KB events, which is why most apps don't handle it. Seems like an issue in Windows itself.

As for multiple instances of this menu existing simultaneously, when a user right-clicks the tray icon, couldn't you do a search for existing windows and close them before opening a new one?

**Comment by steam3d** [comment]
@stamminator Yes, I will close the current window if the user requests a new one.

**Comment by stamminator** [comment]
@steam3d As my recording shows, that isn't always the case. It's easy enough to reproduce that I frequently run into it after opening the Action Center to enable bluetooth and the right-clicking MagicPods to check on its connection status.

![Animation](https://github.com/user-attachments/assets/1c3ef281-83ed-4a61-8f07-6b29f4d52ab0)

Once it's in this state, it stops closing prior windows, and only terminating MagicPods fixes the problem.

**Comment by steam3d** [comment]
@stamminator I know. It's a known issue. As I said above, I cannot globally catch user input due to privacy concerns to fix this.

**Comment by stamminator** [comment]
So, it looks like there are two separate issues here.

The first is that certain states in Windows (what I'm calling BTTF, but there may be others) cause menus opened from tray icons while in this state to stop receiving deactivated events. This results in each app's menu getting "orphaned" until the user once again interacts with the tray icon while no longer in this BTTF state, which I document [here](https://github.com/ramensoftware/windhawk-mods/discussions/1205#discussioncomment-11175614). While it is possible to listen to global M/KB events as a workaround (which might be what Monitorian does, but I'm not sure), this introduces privacy concerns and puts the ball in Microsoft's court to fix.

The second issue is different and I believe a legitimate bug in MagicPods. Multiple tray menu instances should not be able to exist simultaneously. While the Windows bug is a scenario under which this defect can take place, other apps successfully avoid this without global event listening by doing a manual check for existing windows within their app's process and not relying entirely on Windows-emitted deactivate events. Regardless of what caused a window to get orphaned, once the user manually clicks on an app's tray icon again, that app has full control to terminate orphaned windows within its own process without any need to listen to global events. Therefore it is fixable.

**Comment by steam3d** [comment]
@stamminator I will fix the second issue in future updates.

**Comment by steam3d** [comment]
https://github.com/user-attachments/assets/c7a66505-b9da-46e6-875f-e0e5e2914677

On Windows 11 LTSC, the MagicPods context menu correctly detects when it is deactivated, but on my main Windows 11 Pro, I experience the same behavior as you

**Comment by steam3d** [comment]
https://github.com/steam3d/MagicPods-Windows/issues/414

**Comment by stamminator** [comment]
Version 2.0.81 fixed it for me. Thanks!
